src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -771,6 +601,256 @@ internal override bool CallsAreOmitted(SyntaxTree syntaxTree)
+            {
+            }
+
+            internal sealed override SourceOrdinaryMethodSymbol OtherPartOfPartial
+                => null;
+
+            protected sealed override TypeSymbol ExplicitInterfaceType
+                => null;
+
+            protected sealed override MethodSymbol FindExplicitlyImplementedMethod(BindingDiagnosticBag diagnostics)
+                => null;
+
+            public sealed override ImmutableArray<TypeParameterSymbol> TypeParameters
+                => ImmutableArray<TypeParameterSymbol>.Empty;
+
+            public sealed override ImmutableArray<ImmutableArray<TypeWithAnnotations>> GetTypeParameterConstraintTypes()
+                => ImmutableArray<ImmutableArray<TypeWithAnnotations>>.Empty;
+
+            public sealed override ImmutableArray<TypeParameterConstraintKind> GetTypeParameterConstraintKinds()
+                => ImmutableArray<TypeParameterConstraintKind>.Empty;
+
+            protected sealed override void CheckConstraintsForExplicitInterfaceType(ConversionsBase conversions, BindingDiagnosticBag diagnostics)
+            {
+            }
+        }
+
+        /// <summary>
+        /// Specialized subclass of SourceOrdinaryMethodSymbol for less common cases.  Specifically, we only use this
+        /// for methods that are:
+        /// <list type="number">
+        /// <item>Generic.</item>
+        /// <item>An explicit interface implementation.</item>
+        /// <item>Partial.</item>
+        /// </list>
+        /// </summary>
+        private sealed class SourceOrdinaryMethodSymbolComplex : SourceOrdinaryMethodSymbol
+        {
+            private readonly TypeSymbol _explicitInterfaceType;
+
+            private readonly TypeParameterInfo _typeParameterInfo;
+
+            /// <summary>
+            /// If this symbol represents a partial method definition or implementation part, its other part (if any).
+            /// This should be set, if at all, before this symbol appears among the members of its owner.  
+            /// The implementation part is not listed among the "members" of the enclosing type.
+            /// </summary>
+            private SourceOrdinaryMethodSymbol _otherPartOfPartial;
+
+            public SourceOrdinaryMethodSymbolComplex(
+                NamedTypeSymbol containingType,
+                TypeSymbol explicitInterfaceType,
+                string name,
+                Location location,
+                MethodDeclarationSyntax syntax,
+                MethodKind methodKind,
+                bool isNullableAnalysisEnabled,
+                BindingDiagnosticBag diagnostics)
+                : base(containingType, name, location, syntax, methodKind, isNullableAnalysisEnabled, diagnostics)
+            {
+                _explicitInterfaceType = explicitInterfaceType;
+                _typeParameterInfo = new TypeParameterInfo { LazyTypeParameters = MakeTypeParameters(syntax, diagnostics) };
```

Comment:
>new TypeParameterInfo { LazyTypeParameters = MakeTypeParameters(syntax, diagnostics) }; [](http://example.com/codeflow?start=37&length=87)

It looks like we used to reuse singleton object for no-type-parameters case and that optimization got undone.

Reply:
done!